pkg/logs/tekton_logging_test.go

```diff
@@ -803,6 +805,133 @@ func TestGetRunningBuildLogsMetapipelineAndPendingGenerated(t *testing.T) {
 		stripansi.Strip(firstLine), "'app-extension' should be the first stage logged")
 }
 
+func TestGetTektonPipelinesWithFailedAndRetriedPipeline(t *testing.T) {
+	jxClient, tektonClient, _, _, ns := getFakeClientsAndNs(t)
+
+	tl := TektonLogger{
+		JXClient:     jxClient,
+		TektonClient: tektonClient,
+		Namespace:    ns,
+		LogWriter: &TestWriter{
+			StreamLinesLogged: make([]string, 0),
+			SingleLinesLogged: make([]string, 0),
+		},
+		LogsRetrieverFunc: LogsProvider,
+	}
+
+	_, err := jxClient.JenkinsV1().PipelineActivities(ns).Create(&v1.PipelineActivity{
+		ObjectMeta: v12.ObjectMeta{
```

Comment:
I'd try to separate the test setup out a bit. It is hard to narrow down on the actual test(assertion).
The whole set up of the fake client could, for example, happen in a _setUp_ function and the resources itself could be extracted and appropriately named, e.g. `succeededPipelineRun`, `activePipelineRun`.

Reply:
Ok, will do tomorrow.

Reply:
So the `PipelineRun`s still differ too much to go that route. I did tease out the `PA1` `PipelineActivity` creation, but the cases where we create runs rather than just load from the disk are a hassle - if I have a chance today, I'm going to tweak those to load from disk instead.

Reply:
Ok, moved the runs to be loaded from disk. I left the activities as is.